python/lsst/cp/pipe/utils.py

```diff
@@ -540,6 +540,37 @@ def arrangeFlatsByExpTime(exposureList):
     return flatsAtExpTime
 
 
+def arrangeFlatsByExpId(exposureList):
+    """Arrange exposures by exposure ID.
+
+    Parameters
+    ----------
+    exposureList : `list`[`lsst.afw.image.exposure.exposure.ExposureF`]
+        Input list of exposures.
+
+    Returns
+    ------
+    flatsAtExpTime : `dict` [`float`,
+                      `list`[`lsst.afw.image.exposure.exposure.ExposureF`]]
+        Dictionary that groups flat-field exposures that have the same
+        exposure time (seconds).
+    """
+    flatsAtExpId = {}
+    sortedExposures = sorted(exposureList, key=lambda exp: exp.getInfo().getVisitInfo().getExposureId())
+
+    for jPair, exp in enumerate(sortedExposures):
```

Comment:
Could you please a comment explaining a bit better what's happening here? I think I understand, but it would be easier to read.

Reply:
I put it into a Notes section, as I couldn't figure out how to split it into comment sized chunks.